float32 wait_time